File: LAB 5/Scaling_with_Respect_to_the_Origin.py
import matplotlib.pyplot as plt
import numpy as np

ORIGINAL_POINTS = np.array([[2, 2], [5, 2], [5, 5], [2, 5], [2, 2]])
SX, SY = 1.5, 0.8

plt.figure(figsize=(10, 5))

plt.subplot(1, 2, 1)
plt.grid(True)
plt.axis('equal')
plt.title('Original Square')
plt.xlabel('X-axis')
plt.ylabel('Y-axis')
plt.plot(ORIGINAL_POINTS[:, 0], ORIGINAL_POINTS[:, 1], 'b-', linewidth=2)
plt.fill(ORIGINAL_POINTS[:, 0], ORIGINAL_POINTS[:, 1], 'b', alpha=0.3)
plt.xlim(-1, 8)
plt.ylim(-1, 8)
plt.axhline(y=0, color='k', linestyle='-', linewidth=0.5)
plt.axvline(x=0, color='k', linestyle='-', linewidth=0.5)

plt.subplot(1, 2, 2)
plt.grid(True)
plt.axis('equal')
plt.title(f'Scaled: Sx={SX}, Sy={SY}')
plt.xlabel('X-axis')
plt.ylabel('Y-axis')

SCALED_POINTS = ORIGINAL_POINTS * [SX, SY]
plt.plot(SCALED_POINTS[:, 0], SCALED_POINTS[:, 1], 'r-', linewidth=2)
plt.fill(SCALED_POINTS[:, 0], SCALED_POINTS[:, 1], 'r', alpha=0.3)
plt.xlim(-1, 8)
plt.ylim(-1, 8)
plt.axhline(y=0, color='k', linestyle='-', linewidth=0.5)
plt.axvline(x=0, color='k', linestyle='-', linewidth=0.5)

plt.tight_layout()
plt.show()

File: LAB 5/Scaling_with_Respect_to_the_Origin.m
clc; clear; close all;

ORIGINAL_POINTS = [2, 2; 5, 2; 5, 5; 2, 5; 2, 2];
SX = 1.5; SY = 0.8;

figure('Position', [100, 100, 1000, 400]);

subplot(1, 2, 1);
hold on; grid on; axis equal;
title('Original Square');
xlabel('X-axis'); ylabel('Y-axis');
plot(ORIGINAL_POINTS(:, 1), ORIGINAL_POINTS(:, 2), 'b-', 'LineWidth', 2);
fill(ORIGINAL_POINTS(:, 1), ORIGINAL_POINTS(:, 2), 'b', 'FaceAlpha', 0.3);
xlim([-1, 8]); ylim([-1, 8]);
line([-1, 8], [0, 0], 'Color', 'k', 'LineWidth', 0.5);
line([0, 0], [-1, 8], 'Color', 'k', 'LineWidth', 0.5);

subplot(1, 2, 2);
hold on; grid on; axis equal;
title(['Scaled: Sx=', num2str(SX), ', Sy=', num2str(SY)]);
xlabel('X-axis'); ylabel('Y-axis');

SCALED_POINTS = ORIGINAL_POINTS .* [SX, SY];
plot(SCALED_POINTS(:, 1), SCALED_POINTS(:, 2), 'r-', 'LineWidth', 2);
fill(SCALED_POINTS(:, 1), SCALED_POINTS(:, 2), 'r', 'FaceAlpha', 0.3);
xlim([-1, 8]); ylim([-1, 8]);
line([-1, 8], [0, 0], 'Color', 'k', 'LineWidth', 0.5);
line([0, 0], [-1, 8], 'Color', 'k', 'LineWidth', 0.5);
